src/callbacks/display.py

```diff
@@ -355,10 +368,19 @@ def update_heatmap(
         go_live:                n_clicks for live mode button
     Returns:
         fig:                    updated heatmap
+        stats:                  statistics text
     """
     # user select a group of points
     if selected_data is not None and len(selected_data["points"]) > 0:
         selected_indices = [point["pointIndex"] for point in selected_data["points"]]
+        
+        # Check if selection exceeds limit
+        if len(selected_indices) > MAX_HEATMAP_SELECTION:
+            return (
+                plot_empty_heatmap(),
+                f"⚠️ Selection too large ({len(selected_indices)} points). "
```

Comment:
With this check, we would be rejecting the selection of more than 30 points, but I believe this is a problem when using tiled only. If we use the prefect version (not live) with filesystem, I believe we can load more than 30 points. I wonder if perhaps something we could do is the following: when latency issues are detected, we could instead plot a random sample of the selected points - with a [notification](https://github.com/mlexchange/mlex_latent_explorer/blob/6177f91e37187b91c2767f0fc8b7ccf9dd50a7c1/src/utils/plot_utils.py#L517) for the user.
This is not MVP for beamtime, so we could check if we're in live mode as a patch to avoid impacting the offline mode with this merge.